### DISCUSSION TITLE
Setup the [TestHarness] where you compare the node value

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#26662] [Q&A General] [answered]
Hi guys,

I'm very new to `TestHarness` I know only the way suggested in the example (diffusion test)

https://mooseframework.inl.gov/moose/python/TestHarness.html

just wondering if it is possible to do the test where I want to compare and check the nodes value, lets say the test will pass if they are the same (or within 1% different). [this is to always check our nodal constraint object, to check if it's still work with current version of MOOSE]

I have the list of all the node pairs for this test [Below], I'm just not sure how to implement this

masterNd, slaveNd_0
5,	21
6,	22
7,	23
8,	24
13,	25
14,	26
15,	27
16,	28

Please let me know if you have any suggestion.

Thank you!
Traiwit

**Comment by GiudGiud** [comment]
Hello

So you want the test to only a check a few nodal values for a given variable?
If the list is not very long, I would do:
- a [CSVDiff](https://mooseframework.inl.gov/python/testers/CSVDiff_tester.html) test
- with a [NodalValueSampler](https://mooseframework.inl.gov/source/vectorpostprocessors/NodalValueSampler.html) on a custom sideset that contains all the nodes you care about. 

Now for a 1% tolerance, you will want to increate rel_tol from 5e-6 to 1e-2. That parameter goes in the `tests` file, inside the relevant test sub-block

Guillaume

**Reply by Traiwit** [reply, marked as answer]
Hi @GiudGiud 

CSVDiff couldn't do what I want, I want to compare nodes value after the simulation is done

not the .csv in the gold folder and the .csv produced by the test_run

However, I managed to use `RunCommand` with .py script instead and it kinda works

Thank you!
Traiwit
